### PR TITLE
[Federation][init-10c] Fix hyperkube binary path in the image and disable auth mechanisms we don't support for now.

### PR DESCRIPTION
**Pull request by madhusudancs**
Please review only the last commit here. This is based on PR #36047 which will be reviewed independently.

Design Doc: PR #34484

cc @kubernetes/sig-cluster-federation @nikhiljindal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36048)

### CODE REVIEW

**Comment by quinton-hoole** [comment]
LGTM

File: federation/pkg/kubefed/init/init.go [hunk 1]
```diff
@@ -389,8 +389,6 @@ func createAPIServer(clientset *client.Clientset, namespace, name, image, creden
 		"--etcd-servers=http://localhost:2379",
 		"--service-cluster-ip-range=10.0.0.0/16",
 		"--secure-port=443",
-		"--token-auth-file=/etc/federation/apiserver/known_tokens.csv",
-		"--basic-auth-file=/etc/federation/apiserver/basic_auth.csv",
```
Comment (nikhiljindal) [inline federation/pkg/kubefed/init/init.go]: We do support basic auth. That is how I connect to federation apiserver on my browser. Did we break it?

Comment (nikhiljindal) [inline federation/pkg/kubefed/init/init.go]: oh sorry. We do not support these auth mechanisms in the new setup? That is fine for alpha. I assume you have plans to support it later.

Comment (madhusudancs) [inline federation/pkg/kubefed/init/init.go]: Yeah. I have immediate plans to make this work. See the TODO at the top of this file. Also, the code for this is already written in my, now abandoned, PR - #30615 (https://github.com/kubernetes/kubernetes/pull/30615/commits/aac3f0af52a89b5333501b5395b064c3d3e3776f). I just need to copy it over to this package.

**Pull request merged**